File: 프로그래머스/2/42578. 의상/의상.py
# from collections import Counter

# def solution(clothes):
#     answer = 1
    
#     clothes_kind = []
#     for item in clothes:
#         clothes_kind.append(item[1])
#     clothes_kind = Counter(clothes_kind)
    
#     for item in clothes_kind:
#         answer *= (clothes_kind[item]+1)

#     return (answer-1)


def solution(clothes):
    clothes_type = {}
    
    for c, t in clothes:
        if t not in clothes_type:
            clothes_type[t] = 2
        else:
            clothes_type[t] += 1
            
    cnt = 1
    for num in clothes_type.values():
        cnt *= num
        
    return cnt - 1

File: 프로그래머스/2/42578. 의상/의상.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var dic = [String: Int]()
    
    for i in clothes {
        if let type = dic[i[1]] {
            dic[i[1]]? += 1
        } else {
            dic[i[1]] = 1
        }
    }
    
    var res = 1
    dic.forEach { res *= ($0.value+1) }
    return res>0 ? res-1 :  0
}